docker/docker_client.go

```diff
@@ -50,12 +50,19 @@ const (
 var (
 	// ErrV1NotSupported is returned when we're trying to talk to a
 	// docker V1 registry.
-	ErrV1NotSupported = errors.New("can't talk to a V1 docker registry")
-	// ErrUnauthorizedForCredentials is returned when the status code returned is 401
-	ErrUnauthorizedForCredentials = errors.New("unable to retrieve auth token: invalid username/password")
-	systemPerHostCertDirPaths     = [2]string{"/etc/containers/certs.d", "/etc/docker/certs.d"}
+	ErrV1NotSupported         = errors.New("can't talk to a V1 docker registry")
+	systemPerHostCertDirPaths = [2]string{"/etc/containers/certs.d", "/etc/docker/certs.d"}
 )
 
+// ErrUnauthorizedForCredentials is returned when the status code returned is 401
+type ErrUnauthorizedForCredentials struct { // We only use a struct to allow a type assertion, without limiting the contents of the error otherwise.
+	Err error
+}
+
+func (e ErrUnauthorizedForCredentials) Error() string {
```

Comment:
This works, but we might prefer to return only the previous hard-coded string, or an `errors.Wrapf`-like text (“invalid username/password: %s"; “invalid username/password, server reported: %s") — or something similar that works well for typical server-returned error values, including for Quay.

Ideally, the callers should not _have_ to reformat the error string at all.

Reply:
can I use this?
```go
func (e ErrUnauthorizedForCredentials) Error() string {
	return errors.Wrapf(e.Err, "unable to retrieve auth token: invalid username/password").Error()
}
```

Reply:
That’s a very expensive way to say
```go
return fmt.Sprintf("unable to retrieve auth token: invalid username/password: %s", e.Err.Error())
```
Not really blocking, but I don’t see much reason to drag in `Wrapf` into it.